File: ecommerce-api-legacy/tools/compare.js
'use strict';

/** Compara duas execuções do smoke e falha se algum status divergir. */
const fs = require('fs');

const indexar = (caminho) =>
  Object.fromEntries(JSON.parse(fs.readFileSync(caminho, 'utf8')).results.map((r) => [r.name, r]));

const antes = indexar(process.argv[2]);
const depois = indexar(process.argv[3]);
const falhas = [];

for (const [nome, a] of Object.entries(antes)) {
  const d = depois[nome];
  if (!d) falhas.push(`${nome}: ausente depois`);
  else if (a.status !== d.status) falhas.push(`${nome}: ${a.status} -> ${d.status}`);
}

console.log(`${Object.keys(antes).length} endpoints comparados`);
console.log(falhas.length ? `DIVERGENCIAS:\n  ${falhas.join('\n  ')}` : 'PARIDADE OK');
process.exitCode = falhas.length ? 1 : 0;

File: code-smells-project/tools/compare.py
import json, sys
antes = {r['name']: r for r in json.load(open(sys.argv[1]))['results']}
depois = {r['name']: r for r in json.load(open(sys.argv[2]))['results']}
falhas = []
for nome, a in antes.items():
    d = depois.get(nome)
    if d is None:
        falhas.append(f'{nome}: ausente depois')
    elif a['status'] != d['status']:
        falhas.append(f"{nome}: {a['status']} -> {d['status']}")
print(f'{len(antes)} endpoints comparados')
print('PARIDADE OK' if not falhas else 'DIVERGENCIAS:\n  ' + '\n  '.join(falhas))
sys.exit(1 if falhas else 0)
